src/pathogena/util.py

```diff
@@ -271,48 +277,124 @@ def hash_file(file_path: Path) -> str:
+max_concurrent_chunks = int(os.getenv("MAX_CONCURRENT_CHUNKS", 5))
+
+
+def chunk_file(file_path: Path, chunk_size: int) -> Generator[bytes, None, None]:
+    """Splits the file into chunks of the specified size.
+
+    Args:
+        file_path (Path): Path to file being uploded
+        chunk_size (int): Size of file chunks, defaults to 5 mb
+
+    Yields:
+        Generator[bytes,None, None]: generator for getting file chunk
+    """
+    with open(file_path, "rb") as file:
+        while chunk := file.read(chunk_size):
+            yield chunk
+
+
 @retry(wait=wait_random_exponential(multiplier=2, max=60), stop=stop_after_attempt(10))
-def upload_file(
-    sample_id: int,
-    file_path: Path,
+def upload_chunk(
+    batch_pk: int,
     host: str,
     protocol: str,
     checksum: str,
     dirty_checksum: str,
-) -> None:
-    """Upload a file to the server with retries.
+    chunk: bytes,
+    chunk_index: int,
+) -> Response:
+    """Upload a single file chunk.
 
     Args:
-        sample_id (int): The ID of the sample.
-        file_path (Path): The path to the file to be uploaded.
-        host (str): The host server.
-        protocol (str): The protocol to use (e.g., 'http', 'https').
-        checksum (str): The checksum of the file.
-        dirty_checksum (str): The dirty checksum of the file.
+        batch_pk (int): ID of sample to upload
+        host (str): pathogena host, e.g api.upload-dev.eit-pathogena.com
+        protocol (str): protocol, default https
+        checksum (str): sample metadata if decontaminated
+        dirty_checksum (str): sample metadata pre-decontimation
+        chunk (bytes): File chunk to be uploaded
+        chunk_index (int): Index representing what chunk of the whole sample file this chunk is from 0...total_chunks
     """
-    with (
-        httpx.Client(
-            event_hooks=httpx_hooks,
-            transport=httpx.HTTPTransport(retries=5),
-            timeout=7200,  # 2 hours
-        ) as client,
-        open(file_path, "rb") as fh,
-    ):
-        client.post(
-            f"{protocol}://{host}/api/v1/samples/{sample_id}/files",
+    with httpx.Client(
+        event_hooks=httpx_hooks,
+        transport=httpx.HTTPTransport(retries=5),
+        timeout=7200,  # 2 hours
+    ) as client:
+        return client.post(
+            f"{protocol}://{host}/api/v1/batches/{batch_pk}/uploads/upload-chunk/",
             headers={"Authorization": f"Bearer {get_access_token(host)}"},
-            files={"file": fh},
-            data={"checksum": checksum, "dirty_checksum": dirty_checksum},
+            files={"file": chunk},
+            data={
+                "checksum": checksum,
+                "dirty_checksum": dirty_checksum,
+                "chunk_index": chunk_index,
+            },
         )
 
 
+@retry(wait=wait_random_exponential(multiplier=2, max=60), stop=stop_after_attempt(10))
+def upload_file_as_chunks(
```

Comment:
first chunk may fail with 400 because of an invalid mime-type as per a latest feature, see: https://pathogena.atlassian.net/browse/EV-1522.

so i think it is better not to retry unnecessarily if we receive a 400 response especially for the first chunk.

Reply:
Will work on other comments, but for this one should it be part of another ticket more like this one?
https://pathogena.atlassian.net/jira/software/c/projects/EV/boards/9?assignee=712020%3Ace3fabf3-a2a4-4ca4-8601-ce98cfd08508&selectedIssue=EV-1535

Reply:
for admin-app, it was a new change at that time which we chose to implement later.
but for this task, i prefer handling it in one go as we are refactoring the logic just not to revisit and refactor same code one more time.

but it is your call to do it within this PR or to do with a different ticket.

Reply:
No more than happy to work on it for this PR, just though better to get an opinion first since I was already aware of the other ticket, thanks!